features/0434-outofband/README.md

```diff
@@ -0,0 +1,417 @@
+
+The *sender* state machine is as follows:
+
+[![sender state machine google doc](state-machine-sender.png)](https://docs.google.com/spreadsheets/d/1o1szGCMx6FzrxfAKV76IEYmeKRJqvV3gd1r3Qfv1Q9o/edit#gid=1176419697)
+
+Note the "optional" reference under the second event in the `await-response` state. That is to indicate that an out of band message might be a single use message with a transition to done, or reusable message (received by many receivers) with a transition back to `await-response`.
+
+The *receiver* state machine is as follows:
+
+[![receiver state machine google doc](state-machine-receiver.png)](https://docs.google.com/spreadsheets/d/1o1szGCMx6FzrxfAKV76IEYmeKRJqvV3gd1r3Qfv1Q9o/edit#gid=10874521)
+
+Worth noting is the first event of the `done` state, where the receiver may receive the message multiple times. This represents, for example, an agent returning to the same website and being greeted with the same QR code each time.
+
+### Messages
+
+There are two messages in the out of band protocol. They are both sent by the *sender*. Only one of the message types is ever used in an instance of the protocol.
+
+#### Message Type: `https://didcomm.org/outofband/VER/invitation`
+
+```jsonc
+{
+  "@type": "https://didcomm.org/outofband/VER/invitation",
+  "@id": "<id used for context as pthid>",
+  "label": "Faber College",
+  "protocols": [
+      "https://didcomm.org/didexchange/1.0",
+      "https://didcomm.org/connections/1.0"
+      ]
+  "service": ["did:sov:LjgpST2rjsoxYegQDRm7EL"]
+}
+```
+
+The items in the message are:
+
+- `@type` - the DIDComm message type
+- `@id` - the unique ID of the message. The ID should be used as the **parent** thread ID (`pthid`) for the response message, rather than the more common thread ID (`thid`) of the response message. This enables multiple uses of a single `invitation` message.
+- `label` - a self-attested string that the receiver may want to display to the user.
+- `protocols` - array of protocols in the order of preference of the sender that the receiver can use in responding to the message. As implied by the message type of invitation, these are not arbitrary protocols but rather protocols that result in the establishment of a connection.
+- `service` - an item that is the equivalent of the service block of a DIDDoc that the receiver is to use in responding to the message. Additional details below.
+
+While the _receiver_ is expected to respond with an initiating message from the chosen protocol using a specified service, the receiver may respond by reusing an existing connection. Specifically, if a connection they have was created from an out of band `invitation` from the same public DID of a new `invitation`, the receiver **SHOULD** use the existing connection in responding to the `invitation`. The selected `protocol` being used by the receiver may have a specific message type to use for this purpose.
```

Comment:
We've been using the word "recipient" instead of "receiver" for protocols. That's because recipients are usually people, whereas receivers are usually pieces of tech. (In phone tech, the receiver is hardware, the recipient of a call is a person. The recipient of an email message is Fred; the receiver is usually his MTA.) Maybe that distinction is not super important; this isn't a big deal to me, but I wanted to point it out in case you think it's worth doing something about.

Reply:
I tried a few things and my preference is receiver.

Others?